android/TodokTodok/app/src/main/java/com/team/todoktodok/presentation/view/book/vm/SelectBookViewModelFactory.kt

```
@@ -0,0 +1,17 @@
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.team.domain.repository.BookRepository
+
+class SelectBookViewModelFactory(
+    private val bookRepository: BookRepository,
+) : ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        if (modelClass.isAssignableFrom(SelectBookViewModel::class.java)) {
+            return SelectBookViewModel(bookRepository) as T
+        } else {
+            throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)